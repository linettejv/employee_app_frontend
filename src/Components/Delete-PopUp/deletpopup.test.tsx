import { fireEvent, render, screen } from '@testing-library/react';
import DeletePopup from './deletepopup';

jest.mock('../../Pages/Employee/api', () => ({
  useDeleteEmployeeMutation: jest.fn(() => [jest.fn()])
}));

describe('Testing DeletePopup', () => {
  test('Checking snapShots for deltePopup:', () => {
    const element = render(<DeletePopup />);

    expect(element).toMatchSnapshot();
  });
  test('testing handle delete clicks ', () => {
    const mockfn = jest.fn();


    render(<DeletePopup />);

    const element = screen.getByTestId('popup-button-test');

    fireEvent.click(element);

  });
});
