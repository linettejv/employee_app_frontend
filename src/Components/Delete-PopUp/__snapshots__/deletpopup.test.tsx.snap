// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DeletePopup Checking snapShots for deltePopup: 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="overlay"
      >
        <div
          class="modal"
        >
          <img
            class="close"
            data-testid="popup-img-test"
            src="/assets/img/close.png"
          />
          <h6>
            Are you sure ?
          </h6>
          <p>
            The Employee Details will be removed from the database
          </p>
          <div
            class="button-class"
          >
            <button
              class="button1"
            >
              Confirm
            </button>
            <button
              class="button2"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="overlay"
    >
      <div
        class="modal"
      >
        <img
          class="close"
          data-testid="popup-img-test"
          src="/assets/img/close.png"
        />
        <h6>
          Are you sure ?
        </h6>
        <p>
          The Employee Details will be removed from the database
        </p>
        <div
          class="button-class"
        >
          <button
            class="button1"
          >
            Confirm
          </button>
          <button
            class="button2"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
